zuul-core/src/main/java/com/netflix/netty/common/proxyprotocol/HAProxyMessageChannelHandler.java

```diff
@@ -63,8 +62,11 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                             throw new IllegalArgumentException("unknown proxy protocl" + destinationAddress);
                         case TCP4:
                         case TCP6:
-                            addr = new InetSocketAddress(
+                            InetSocketAddress inetAddr = new InetSocketAddress(
                                     InetAddresses.forString(destinationAddress), hapm.destinationPort());
+                            addr = inetAddr;
+                            // setting PPv2 explicitly because SourceAddressChannelHandler.ATTR_LOCAL_ADDR could be PPv2 or not
+                            channel.attr(SourceAddressChannelHandler.ATTR_PROXY_PROTOCOL_DESTINATION_ADDRESS).set(inetAddr);
```

Comment:
My point was this line isn't really unit tested.
Perhaps, it's a good opportunity to adding tests for `HAProxyMessageChannelHandler`.